src/service/viewer-impl.js

```diff
@@ -661,12 +675,29 @@ export class ViewerImpl {
       observable = new Observable();
       this.messageObservables_[eventType] = observable;
     }
-    return observable.add(handler);
+    const unlistenFn = observable.add(handler);
+    if (this.receivedMessageQueue_[eventType]) {
+      this.receivedMessageQueue_[eventType].forEach((message) => {
+        observable.fire(message.data);
+        message.deferred.resolve();
+      });
+      this.receivedMessageQueue_[eventType] = [];
+    }
+    return unlistenFn;
   }
 
   /** @override */
   onMessageRespond(eventType, responder) {
     this.messageResponders_[eventType] = responder;
+    if (this.receivedMessageQueue_[eventType]) {
+      this.receivedMessageQueue_[eventType].forEach((message) => {
+        responder(message.data).then(
```

Comment:
Responders are allowed to return `undefined`, so this will result in an error. Instead, you can just resolve with the returned value:

```js
message.deferred.resolve(responder(message.data))
```